src/main/java/seedu/recipe/model/cooked/UniqueRecordList.java

```diff
@@ -154,7 +151,8 @@ private void setGoalsTally() {
                 }
             }
         }
-        internalGoalsList.addAll(new GoalCount(new Goal("Herbivore"), goalMap.get("Herbivore")),
+
+        internalGoalsList.setAll(new GoalCount(new Goal("Herbivore"), goalMap.get("Herbivore")),
```

Comment:
Changed `addAll()` to `setAll()` to mimic clear() -> addAll(), hence skipping the need for an `isEmpty()` check.